cpp/src/inflight_requests.cpp

```diff
@@ -12,31 +12,42 @@ namespace ucxx {
 
 InflightRequests::~InflightRequests() { cancelAll(); }
 
-size_t InflightRequests::size() { return _trackedRequests->_inflight->size(); }
+size_t InflightRequests::size()
+{
+  std::scoped_lock localLock{_mutex};
```

Comment:
This mutex is protecting `_trackedRequests` being rewritten out from under us, and the lock on `->_mutex` is protecting concurrent modification?

Reply:
`InflightRequests::_mutex` (used with what I called `localLock`) is a reentrant lock to prevent two threads from attempting to modify resources (`_trackedRequests`) simultaneously. Reentrancy is required for cases where `cancelAll()` ends up triggering `remove()` (if the object's refcount goes to 0), but we still require exclusive access to modify the actual request trackers separately (inflight/canceling requests) depending on the operation.

Reply:
Thanks!